x-pack/legacy/plugins/infra/public/components/logging/log_text_stream/loading_item_view.tsx

```diff
@@ -28,122 +44,202 @@ export class LogTextStreamLoadingItemView extends React.PureComponent<
+  'position' | 'rangeEdge' | 'onExtendRange' | 'onStreamStart'
+>> = ({ position, rangeEdge, onExtendRange, onStreamStart }) => {
+  if (rangeEdge === 'now' && position === 'end') {
+    return (
+      <EuiButton onClick={onStreamStart} size="s">
+        <FormattedMessage id="xpack.infra.logs.streamLive" defaultMessage="Stream live" />
+      </EuiButton>
+    );
+  }
+
+  const iconType = position === 'start' ? 'arrowUp' : 'arrowDown';
+
+  if (!rangeEdge) {
+    return null;
+  }
 
-const ProgressEntry: React.FC<ProgressEntryProps> = props => {
-  const { alignment, children, className, color, isLoading } = props;
+  const extendedRange = extendDatemath(rangeEdge, position === 'start' ? 'before' : 'after');
 
-  // NOTE: styled-components seems to make all props in EuiProgress required, so this
-  // style attribute hacking replaces styled-components here for now until that can be fixed
-  // see: https://github.com/elastic/eui/issues/1655
-  const alignmentStyle =
-    alignment === 'top' ? { top: 0, bottom: 'initial' } : { top: 'initial', bottom: 0 };
+  if (!extendedRange || !('diffUnit' in extendedRange)) {
+    return null;
+  }
 
   return (
-    <ProgressEntryWrapper className={className}>
-      <EuiProgress
-        style={alignmentStyle}
-        color={color}
-        size="xs"
-        position="absolute"
-        {...(!isLoading ? { max: 1, value: 1 } : {})}
-      />
-      {children}
-    </ProgressEntryWrapper>
+    <EuiButton
+      onClick={() => {
+        if (typeof onExtendRange === 'function') {
+          onExtendRange(extendedRange.value);
+        }
+      }}
+      iconType={iconType}
+      size="s"
+    >
+      <ProgressExtendMessage amount={extendedRange.diffAmount} unit={extendedRange.diffUnit} />
+    </EuiButton>
   );
 };
 
-const ProgressEntryWrapper = euiStyled.div`
-  align-items: center;
-  display: flex;
-  min-height: ${props => props.theme.eui.euiSizeXXL};
-  position: relative;
-`;
-
-const ProgressMessage = euiStyled.div`
-  padding: 8px 16px;
-`;
+const ProgressExtendMessage: React.FC<{ amount: number; unit: Unit }> = ({ amount, unit }) => {
+  switch (unit) {
+    case 'ms':
+      return (
+        <FormattedMessage
+          id="xpack.infra.logs.extendTimeframe.ms"
+          defaultMessage="Extend timeframe {amount} milliseconds"
+          values={{ amount }}
+        />
+      );
+    case 's':
+      return (
+        <FormattedMessage
+          id="xpack.infra.logs.extendTimeframe.s"
+          defaultMessage="Extend timeframe {amount} seconds"
+          values={{ amount }}
+        />
+      );
+    case 'm':
+      return (
+        <FormattedMessage
+          id="xpack.infra.logs.extendTimeframe.m"
+          defaultMessage="Extend timeframe {amount} minutes"
+          values={{ amount }}
+        />
+      );
+    case 'h':
+      return (
+        <FormattedMessage
+          id="xpack.infra.logs.extendTimeframe.h"
+          defaultMessage="Extend timeframe {amount} hours"
```

Comment:
<img width="338" alt="Screen Shot 2020-02-19 at 1 35 24 PM" src="https://user-images.githubusercontent.com/1445834/74868962-b5a68400-531c-11ea-8b21-d49c6ff9c8d7.png">

Need to handle the case for `1`

Reply:
Done!